share/gruvbox_config.py

```diff
@@ -0,0 +1,187 @@
+from barpyrus import hlwm
+from barpyrus import widgets as W
+from barpyrus.core import Theme
+from barpyrus import lemonbar
+from barpyrus import conky
+from barpyrus.colors import (
+    AQUA_LIGHT,
+    GREEN_LIGHT,
+    YELLOW_LIGHT,
+    PURPLE_LIGHT,
+    BLUE_LIGHT,
+    ORANGE_LIGHT,
+    RED_LIGHT,
+    FG,
+    BG,
+    col_fmt,
+)
+import sys
+# Copy this config to ~/.config/barpyrus/config.py
+
+# set up a connection to herbstluftwm in order to get events
+# and in order to call herbstclient commands
+hc = hlwm.connect()
+
+# get the geometry of the monitor
+monitor = sys.argv[1] if len(sys.argv) >= 2 else 0
+(x, y, monitor_w, monitor_h) = hc.monitor_rect(monitor)
+height = 14 # height of the panel
+width = monitor_w # width of the panel
+hc(['pad', str(monitor), str(height)]) # get space for the panel
+
+# Conky setup
+custom = ''
+
+mail_symb = '\ue1a8'
+inbox = '/home/axel/.mail/INBOX'
+mail = col_fmt(AQUA_LIGHT) + mail_symb + col_fmt(FG) + ' ${new_mails %s}/${mails %s}' % (inbox, inbox)
+
+disk_symb = '\ue1bb'
+disk_usage = col_fmt(GREEN_LIGHT) + disk_symb + col_fmt(FG) + '${fs_free}/${fs_size}'
+
+on_symb = '\ue10c'
+uptime = col_fmt(GREEN_LIGHT) + on_symb + col_fmt(FG) + '${uptime_short}'
+
+##########
+# volume #
+##########
+vol_off_symb = col_fmt(FG) + '\ue202'
+speaker_symb = col_fmt(YELLOW_LIGHT) + '\ue203'
+headphone_symb = col_fmt(YELLOW_LIGHT) + '\ue0fd'
+bluetooth_symb = col_fmt(BLUE_LIGHT) + '\ue1b5'
+vol_on_symb = '${if_match \"$pa_sink_active_port_name\" == \"analog-output-headphones\"}'
+vol_on_symb += headphone_symb
+vol_on_symb += '${else}'
+
+vol_on_symb += '${if_match \"$pa_sink_active_port_name\" == \"headset-output\"}'
+vol_on_symb += headphone_symb + bluetooth_symb
+vol_on_symb += '${else}'
+vol_on_symb += speaker_symb
+vol_on_symb += '${endif}'
+
+vol_on_symb += '${endif}'
+
+volume = '${if_pa_sink_muted}' + vol_off_symb + '${else}' + vol_on_symb + '${endif}'
+volume += col_fmt(FG) + '${pa_sink_volume}%'
+
+###########
+# various #
+###########
+cpu_symb = '\ue026'
+cpu = col_fmt(PURPLE_LIGHT) + cpu_symb + col_fmt(FG) + '${cpu}%'
+ram_symb = '\ue021'
+ram = col_fmt(PURPLE_LIGHT) + ram_symb + col_fmt(FG) + '${memperc}%'
+net_down_speed_symb = '\ue13c'
+net_down_speed = col_fmt(RED_LIGHT) + net_down_speed_symb + col_fmt(FG) + '${downspeedf}K'
+net_up_speed_symb = '\ue13b'
+net_down_speed = col_fmt(GREEN_LIGHT) + net_up_speed_symb + col_fmt(FG) + '${upspeedf}K'
+
+########
+# wifi #
+########
+wifi_icons = ['\ue218', '\ue219', '\ue21a']
+wifi_delta = 100 / len(wifi_icons)
+wifi_qual = '${wireless_link_qual_perc wlp0s20f3}'
```

Comment:
Can we get rid of this `wlp0s20f3` to make it work for others? In my personal barpyrus config I'm simply listing all network devices: https://github.com/t-wissmann/dotfiles/blob/master/config/barpyrus/config.py#L38..L62

Reply:
I'll check how you do it

Reply:
I now use the first entry in /sys/class/net which starts with 'w' and if none exists I show a disconnected symbol